### DISCUSSION TITLE
airflow rotate-fernet-key raise InvalidToken cryptography.fernet.InvalidToken

### DISCUSSION THREAD

**Discussion by Alaeddine22** [apache/airflow#29700] [Q&A]
Hello,
 I'm having the same issue as mentioned here, i'm pretty sure the old fernetkey is correct :
https://stackoverflow.com/questions/63695273/airflow-rotating-encryption-keys

i traced the error and found that the function **_verify_signature** in (https://github.com/pyca/cryptography/blob/1af039f5fe1ac2710dfe21f6e0b7bafeaa7116eb/src/cryptography/fernet.py#L129) fails at this level **h.verify(data[-32:])** with the error **Signature did not match digest**

airflow 2.5.0
python 3.8
cryptography==38.0.3

any help is appreciated

**Comment by potiuk** [comment]
Maybe whitespace in your key ?

**Reply by Alaeddine22** [reply]
i made sure there are no trailing whitespaces

**Reply by potiuk** [reply]
No idea. I suggest to try to write snippets of Python code that read/verify the fernet key (extracting it from Airflow code and experimenting with it - maybe this is a problem with version of cryptography or smth else). Also if the fernet key is "good" then you should be able to see it working. Also maybe you can try try explain your problem in detail, because it is hard to guess what really is your problem. 

The issue you linked to explains that the user does not know how to do rotation, the answer explains it in detail. So I am not sure what is "the same issue" you mentioned (because seems that issue has an accepted answer).

If you need help, I suggest you explain your situation i more datail: 

* what was the starting point (how yoy are sure your Fernet key is correct) - which version of Airflow it works with (does it still work?) etc. etc.
* what are you trying to achieve
* what exactly (with snippets of code) you attempted to do
* how it did not work for you, what errors you had and at which stage

Without all that detailed information is next to impossible to help you with your problem.

**Reply by Alaeddine22** [reply]
Hello, 
Sorry for the late reply.
I did run the code snippets of python and it didn't work.
I downgraded my cryptography version to the one mentioned here (https://raw.githubusercontent.com/apache/airflow/constraints-2.5.0/constraints-3.7.txt) and didn't fix my problem either.

I ended up with having to TRUNCATE the table CONNECTION and recreating my connections from the airflow front.
PS: Having everything set up, driven by curiosity, I launched the command **rotate-fernet-key** and it failed.

**Reply by potiuk** [reply]
> I launched the command rotate-fernet-key and it failed.

How did you run it and how it failed?
